src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/locations/LocationsReconciliationService.kt

```diff
@@ -249,39 +249,38 @@ class LocationsReconciliationService(
   }
 
   fun historyDoesNotMatch(nomis: LocationResponse, dps: Location): Boolean {
-    if (expectedDpsHistorySize(nomis.amendments) == (dps.changeHistory?.size ?: 0)) {
-      return false
+    val filteredAmendmentResponses = filterAmendmentResponses(nomis.amendments)
+    if ((filteredAmendmentResponses?.size ?: 0) != (dps.changeHistory?.size ?: 0)) {
+      return true
     }
-    val nomisListConverted = nomis.amendments!!.map { a ->
+    val nomisListConverted = filteredAmendmentResponses!!.map { a ->
       ChangeHistory(toHistoryAttribute(a.columnName), a.amendedBy, a.amendDateTime, a.oldValue, a.newValue)
     }
     val nomisSetSorted = nomisListConverted.sortedBy { it.amendedDate + it.attribute + it.amendedBy + it.oldValue + it.newValue }
     val dpsSetSorted = dps.changeHistory!!.sortedBy { it.amendedDate + it.attribute + it.amendedBy + it.oldValue + it.newValue }
 
-    var dpsIndex = 0
-    nomisSetSorted.forEach { nomisItem ->
-      val dpsItem = dpsSetSorted.elementAtOrNull(dpsIndex)
-      if (dpsItem != null) {
-        if (nomisItem.attribute == dpsItem.attribute &&
-          nomisItem.amendedBy == dpsItem.amendedBy &&
-          nomisItem.amendedDate == dpsItem.amendedDate &&
-          nomisItem.oldValue == dpsItem.oldValue &&
-          nomisItem.newValue == dpsItem.newValue
-        ) {
-          // matches
-        } else {
-          return true
-        }
+    if (dpsSetSorted.size > nomisSetSorted.size) {
+      return true
+    }
```

Comment:
Minor style thing for next time, the Kotlin way would read slightly better (might need reverse logic, e.g NOT equal)

`nomisSetSorted.zip(dpsSetSorted).any { (nomisItem, dpsItem) -> 
       nomisItem.attribute == dpsItem.attribute &&
        nomisItem.amendedBy == dpsItem.amendedBy &&
        nomisItem.amendedDate == dpsItem.amendedDate &&
        nomisItem.oldValue == dpsItem.oldValue &&
        nomisItem.newValue == dpsItem.newValue
      ) 
    }`

Reply:
Good idea - done